File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        Map<Integer, Integer> map = new HashMap<>();
        ListNode prec = null;
        ListNode copy = head;
        while(head != null){
            if(!map.containsKey(head.val)){
                map.put(head.val, 1);
            }else{
                while( head != null && map.containsKey(head.val)){head = head.next;}
                map.put(head != null ? head.val : 0, 1);
                prec.next = head;
            }
            prec = head;
            head = head != null ? head.next : null;
        }
        return copy;
    }
}

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        elem = dict()
        prec = None
        copy = head
        while head:
            if head.val not in elem:
                elem[head.val] = 1
            else:
                while head and head.val in elem:
                    head = head.next
                elem[head.val if head != None else 0] = 1
                prec.next = head
            
            prec = head
            head = head.next if head != None else None
        
        return copy
                